src/main/java/net/snowflake/ingest/utils/Constants.java

```diff
@@ -55,11 +56,65 @@ public class Constants {
   public static final String OPEN_CHANNEL_ENDPOINT = "/v1/streaming/channels/open/";
   public static final String REGISTER_BLOB_ENDPOINT = "/v1/streaming/channels/write/blobs/";
 
-  public static enum WriteMode {
+  public enum WriteMode {
     CLOUD_STORAGE,
     REST_API,
   }
 
+  /** Thw write mode to generate Arrow BDEC file. */
+  public enum ArrowBatchWriteMode {
+    /** Stream format is produced by {@link org.apache.arrow.vector.ipc.ArrowStreamWriter}. */
+    STREAM,
+
+    /**
+     * File format is produced by {@link org.apache.arrow.vector.ipc.ArrowFileWriter}.
+     *
+     * <p>The file format is same as stream format but it adds a footer at the end of the file.
+     * The footer contains metadata for quick random access of certain column data in batches
+     * when it is being read on server side. This way there is no need to download and parse the whole file
+     * if only certain columns are requested.
+     */
+    FILE,
+  }
+
+  /** Thw write mode to generate Arrow BDEC file. */
+  public enum BdecVerion {
+    /** Uses Arrow to generate BDEC chunks with {@link ArrowBatchWriteMode#STREAM}. */
+    ONE(1),
+
+    /** Uses Arrow to generate BDEC chunks with {@link ArrowBatchWriteMode#FILE}. */
+    TWO(2);
```

Comment:
nit: could we start from 0 instead of 1? I feel like most CS engineers like zero-based offset :)